guikit/threads.py

```diff
@@ -169,7 +192,10 @@ def run_thread(
         thread = WorkerThread(target, on_abort, on_complete, on_error, daemon)
         thread.connect_events(self._window)
         thread.start()
-        self._workers[thread.ident] = thread
+
+        with self._workers_lock:
```

Comment:
If done this way, the thread might attempt to access the workers dict before it is locked. Shouldn't be `thread.start()` within the locked block?

Reply:
Oops! Yes you're totally right. We should only release the lock once we've added the thread to ``_workers``.